crates/ink/codegen/src/generator/metadata.rs

```diff
@@ -114,6 +116,7 @@ impl Metadata<'_> {
             .module()
             .impls()
             .flat_map(|item_impl| item_impl.iter_constructors())
+            .filter(|constructor| !is_conditionally_excluded(constructor.attrs()))
```

Comment:
I reckon if you encapsulate this filter inside `iter_constructors` and `iter_messages` you could get rid of a lot of these `filter` expressions

Reply:
I was thinking about that, but I don't think it is a good idea because there might be a case when we want to get all messages no matter whether they have `cfg` attributes or not

Reply:
I believe for all current usages that is not the case though.

If we are sticking to conditional compilation semantics then if `cfg` predicates not there it would be as if the code is not there at all.